agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -140,7 +146,7 @@ def apply
     end
 
     # @return [Docker::Container, nil]
-    def ensure_desired_state
+    def ensure_desired_state(restart: false)
```

Comment:
Could we set desired state to `service_pod`? This looks a bit hack-ish.

Reply:
That's kind of the problem, it's not a `desired_state: restart`... if that `@service_pod` gets applied twice, does it restart twice?

It could reset the `desired_state` back to `running` after that goes through, but that's IMO just even more hackish than passing a function parameter?

Reply:
Fair enough but because `restart` is not actually a desired state should we do restart via other method than `ensure_desired_state`?

Reply:
WiP because the container health check worker changes also need something very similar to trigger restarts.